### PR TITLE
[ReversingLabs TitaniumCloud] v2.3.1

### PR DESCRIPTION
**Pull request by MislavReversingLabs**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/ReversingLabs_Titanium_Cloud/Integrations/ReversingLabsTitaniumCloudv2/ReversingLabsTitaniumCloudv2.yml [hunk 1]
```diff
@@ -311,9 +311,27 @@ script:
       default: true
       required: true
     outputs:
+    - contextPath: File.MD5
+      description: MD5 hash.
+      type: Unknown
```
Comment (MichaelYochpaz) [inline Packs/ReversingLabs_Titanium_Cloud/Integrations/ReversingLabsTitaniumCloudv2/ReversingLabsTitaniumCloudv2.yml:316]: Why `Unknown` here? Shouldn't it be:
```suggestion
      type: String
```

Same for some of the other fields (`File.SHA1`, `File.SHA256`)

**Review by MichaelYochpaz — COMMENTED** [review]
Hey, thank you for the contribution :)
Looks good overall from my side of the review, just a small note.

Also, note that [there's a failing validation for the newly added context paths being missing from the README](https://app.circleci.com/pipelines/github/demisto/content/302125/workflows/d554b5a5-495a-4e66-b2ac-9d4565ee31d5/jobs/719272), [and a few unit-tests that fail](https://app.circleci.com/pipelines/github/demisto/content/302125/workflows/d554b5a5-495a-4e66-b2ac-9d4565ee31d5/jobs/719273). Please fix.

File: Packs/ReversingLabs_Titanium_Cloud/Integrations/ReversingLabsTitaniumCloudv2/ReversingLabsTitaniumCloudv2.yml (hunk 1 above)
Comment (MislavReversingLabs) [reply Packs/ReversingLabs_Titanium_Cloud/Integrations/ReversingLabsTitaniumCloudv2/ReversingLabsTitaniumCloudv2.yml:316]: Thank you! I am aware of the failing checks and I am currently resolving them.

**Comment by melamedbn** [comment]
Hi @MislavReversingLabs,

I hope this finds you well. I've reviewed the **Malware Triage - ReversingLabs TitaniumCloud** playbook and have a few suggestions that can enhance its functionality and usability:

1. **Flexibility**: In task 4, it would be helpful to replace the hardcoded input with a playbook input, enabling greater adaptability for the end user.
2. **Task Naming**: For clarity, could we assign meaningful names to tasks 5, 21, 22, and 23?
3. **Context Use**: Based on the attached screenshot, consider opting for a complex context over a simple one for tasks: 5, 19, 20, 21, 22, and 23.
image

4. **Handling Multiple Files**: If there are multiple files in an incident, it's often beneficial to filter using `File.EntryID` where `File.SHA1` matches `inputs.hash`.
5. **Consistency with Inputs**: I noticed a mix of hardcoded values and playbook inputs (like 'hash'). It would be efficient to utilize playbook inputs consistently.
6. **Task Dependency**: To ensure smooth operation, perhaps we could add a generic-polling command to task 23, ensuring it only executes after task 22 completes.
7. **Command Selection**: As per the attached screenshot, could we switch to generic commands in certain tasks instead of instance-specific ones?
Generic
image
Specific
image

8. **Input Description Clarity**: If the commands support only SHA1, we should adjust the description for the 'hash' input. Stating that it supports md5, sha1, and sha256 might be misleading.

Your efforts and contributions are genuinely appreciated. Implementing these suggestions would further refine the playbook. Thank you for your dedication and hard work!

Warm regards,

Ben

**Comment by melamedbn** [comment]
Hi @MislavReversingLabs,

Based on the following feedback:

```
1. Flexibility: In task 4, it would be helpful to replace the hardcoded input with a playbook input, enabling greater adaptability for the end user.
Now I set it to inputs.hash  - is that better?
2. Task Naming: For clarity, could we assign meaningful names to tasks 5, 21, 22, and 23?
Done
3. Context Use: Based on the attached screenshot, consider opting for a complex context over a simple one for tasks: 5, 19, 20, 21, 22, and 23.
Since I don't see any filters and transformers needed in this playbook, is a complex context required?
4. Handling Multiple Files: If there are multiple files in an incident, it's often beneficial to filter using File.EntryID where File.SHA1 matches inputs.hash.
How and where do i do the filtering? I'm not sure what you are refering to.
5. Consistency with Inputs: I noticed a mix of hardcoded values and playbook inputs (like 'hash'). It would be efficient to utilize playbook inputs consistently.
Understood. I'm actually correcting that right now.
6. Task Dependency: To ensure smooth operation, perhaps we could add a generic-polling command to task 23, ensuring it only executes after task 22 completes.
Already being implemented as we speak. However, can you advise me how would you do it in our case? I never used generic polling.
7. Command Selection: As per the attached screenshot, could we switch to generic commands in certain tasks instead of instance-specific ones?
Is the "(ReversingLabs TitaniumCloud v2)" part really the instance name? I always thought that this part of the name says we are using the v2 TitaniumCloud integration instead of the old one (that doesn't even have a lot of the commands I'm using here).
8. Input Description Clarity: If the commands support only SHA1, we should adjust the description for the 'hash' input. Stating that it supports md5, sha1, and sha256 might be misleading.
Not really. The dynamic analysis commands support only SHA-1 but other ReversingLabs commands in this playbook support all three hash types. What should I do in this case?
```
1. Great
3. Required to make the keys usage safer.
4. Are there incidents where their might be multiple files?
6. Sure, You can take a look on Cortex XDR - Alerts Handling and if you won't understand how to implement I'm here for any other quesitons.
7. Yes, please just use the generic command instead the one that specifies the instance name for cases where the user changes the default instance name.
8. If an input is being used for different hashes it's a problem. we better do 3 different inputs (md5, sha1 and sha256) and use them when needed.

Best regards,
Ben

**Comment by MislavReversingLabs** [comment]
>1.  Great
> 2. Required to make the keys usage safer.
> 3. Are there incidents where their might be multiple files?
> 4. Sure, You can take a look on Cortex XDR - Alerts Handling and if you won't understand how to implement I'm here for any other quesitons.
> 5. Yes, please just use the generic command instead the one that specifies the instance name for cases where the user changes the default instance name.
> 6. If an input is being used for different hashes it's a problem. we better do 3 different inputs (md5, sha1 and sha256) and use them when needed.

3. Not really, no.
5. Changing it now.
6. I changed the input hash description SHA1.

**Review by melamedbn — APPROVED** [review]

**Review by MichaelYochpaz — CHANGES_REQUESTED** [review]
@MislavReversingLabs Please address my comment here and then we can merge it.
https://github.com/demisto/content/pull/29609/files#r1322770054

Also, make sure to update the types on the README too.

**Review by MichaelYochpaz — APPROVED** [review]
Looks good, good job :)

**Pull request merged**